### ISSUE TITLE
Set BrowserWindow options defaults for child windows ?

### ISSUE THREAD

**Issue by Gregg8** [atom/electron#2781]
We have an Electron app which launches a new window containing a Flash application from the main window. We set `plugins` to `true` in the `BrowserWindow ` options and away it goes. All good.

However, the Flash app itself then launches another window containing a Flash component. The Flash component fails because it cannot find the Flash plugin. 

This child app works fine if we just launch it directly in a browser OR if we launch it directly from Electron with `plugins` set to `true`.

Could Electron be modified to either:
a) allow these child windows to inherit the options of the parent window, or...
b) allow option defaults to be specified for these child windows

or have I missed something and there's an easy way to currently do this?

Thanks.

**Comment by etiktin** [comment]
This issue seems to be related to #2771 (same solution).

**Comment by Gregg8** [comment]
Thanks for finding the link @etiktin. In that case, the app still works.

In my case, the app becomes unusable, so I don't consider this to be an enhancement , as per the tagging of the other issue.

The inheritance idea would be my preferred solution.

**Comment by zcbenz** [comment]
> However, the Flash app itself then launches another window containing a Flash component. The Flash component fails because it cannot find the Flash plugin.

How does the Flash app open a new window? Does it call `window.open`? If so then it is a duplicate of #2771.

**Comment by Gregg8** [comment]
The Flash app is written in ActionScript3 and the code to open a url in a new window looks something like this:

```
var request = new URLRequest("http://www.url-which-requires-flash.com");
navigateToURL(request, "_blank");
```

The rest is internal mechanics. Perhaps it eventually maps to `window.open`, I don't know.

But I would be grateful for a fix or a workaround to allow windows launched by Electron to then launch windows which have access to the same plugins the Electron windows have access to. How can I do that?

**Comment by zcbenz** [comment]
> But I would be grateful for a fix or a workaround to allow windows launched by Electron to then launch windows which have access to the same plugins the Electron windows have access to. How can I do that?

It will be solved once #2771 gets implemented, I'm closing this as a duplicate.
